### ISSUE TITLE
Document the current state of the fuzzer

### ISSUE THREAD

**Issue by luke-goddard** [tlsfuzzer/tlsfuzzer#726]
Hi, I really like this project but as an outsider, I found the project a little confusing. Your README.md says 'Fuzzer and test suite' and your documentation only talks about the test suite. By looking through the code I would assume the fuzzer is only partially implemented. It would save users a bit of time to mention the state of the fuzzer. I'm quite interested in the project (specifically the fuzzer) and might be a future contributor.

**Comment by tomato42** [comment]
so, the fuzzer part both is and isn't implemented :)

it's implemented in that there are ways for modifying sent messages: https://tlsfuzzer.readthedocs.io/en/latest/modifying-messages.html#modifying-messages fuzzer-like

but it isn't implemented in that the way those modifiers are usually used is through exhaustive iteration; for example, if we have a 2048 bit RSA signature, we flip every bit of the plaintext and see if the message is rejected, then when it's used in the test case, we may run just few random examples of that full list, but we generate the full list:
https://github.com/tomato42/tlsfuzzer/blob/5b32de56839a1c5857e698789c55ce5d9ad6d3fa/scripts/test-tls13-certificate-verify.py#L533-L571
only combinatorial explosion stops us from testing more than 1 flipped bit at a time: if we would generate those modifications on the fly (which is possible, they are pre-generated so that we can deterministically run them or skip them) we could have a script that keeps fuzzing the signature with more and more complex patterns till the end of time :)

to sum it up: if you want to use it as a fuzzer, you totally can, but I limit that kind of use and instead try to create custom tailored test cases, as those are more likely to find issues in a cryptographic protocol like TLS and use fuzzing-like behaviour just to speed up execution

I'm glad you find it interesting enough to consider contributing, feel free to ask questions, I'll gladly discuss solutions or provide explanations for the behaviour; I'm also open to any suggestions to improve documentation, I know it's not in the best shape, but it's a priority for me to fix it

**Comment by tomato42** [comment]
@luke-goddard do you have any follow-up questions?
